src/uint/boxed/encoding.rs

```diff
@@ -131,6 +132,42 @@ impl BoxedUint {
 
         out.into()
     }
+
+    /// Create a new [`BoxedUint`] from the provided big endian hex string.
+    pub fn from_be_hex(hex: &str, bits_precision: u32) -> CtOption<Self> {
+        let nlimbs = (bits_precision / Limb::BITS) as usize;
+        let bytes = hex.as_bytes();
+
+        assert!(
+            bytes.len() == Limb::BYTES * nlimbs * 2,
+            "hex string is not the expected size"
+        );
+
+        let mut res = vec![Limb::ZERO; nlimbs];
+        let mut buf = [0u8; Limb::BYTES];
+        let mut i = 0;
+        let mut err = 0;
+
+        while i < nlimbs {
+            let mut j = 0;
+            while j < Limb::BYTES {
+                let offset = (i * Limb::BYTES + j) * 2;
+                let (result, byte_err) =
+                    encoding::decode_hex_byte([bytes[offset], bytes[offset + 1]]);
+                err |= byte_err;
+                buf[j] = result;
+                j += 1;
+            }
+            res[nlimbs - i - 1] = Limb(Word::from_be_bytes(buf));
+            i += 1;
+        }
+
+        if err == 0 {
+            CtOption::new(Self { limbs: res.into() }, Choice::from(1))
+        } else {
+            CtOption::new(Self::zero_with_precision(bits_precision), Choice::from(0))
+        }
```

Comment:
```suggestion
        CtOption::new(Self { limbs: res.into() }, Choice::from((err == 0) as u8)
```

There may be a better way to construct the `Choice`, though I don't remember the possible values of `err` offhand